src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+
+	const refScrolled = useRef();
+	const refHeader = useRef();
+
+	const [tableHeight, setHeight] = useState(41);
+	const [scrolled, setScrolled] = useState(0);
+
+	const [selectedNumber, tableSelectionStatus, selectionMethods] = useTableSelection(
+		rows,
+	);
 
-	// Table props
+	const classes = useStyles({
+		withoutTopBorder,
+		selectMode,
+		stickyHeader,
+		scrolled,
+		tableHeight,
+		onRowClick,
+	});
 
-	// Table body props
+	useEffect(() => {
+		const handleResize = () => {
+			if (refScrolled.current.offsetHeight < refScrolled.current.scrollHeight)
+				setScrolled(refScrolled.current.offsetWidth - refScrolled.current.clientWidth);
+			else setScrolled(0);
+		};
 
-	// Table cell props
+		handleResize();
 
-	// Table container props
+		window.addEventListener("resize", handleResize);
 
-	// Table footer props
+		return () => window.removeEventListener("resize", handleResize);
+	}, [refScrolled, contentTimestamp]);
 
-	// Table head props
+	useEffect(() => {
+		if (stickyHeader) {
+			setHeight(refHeader.current.clientHeight);
+		}
+	}, [refHeader, stickyHeader]);
 
-	// Table row props
+	const tableHeaders = buildTableHeaders(
+		headers,
+		classes,
+		selectMode,
+		tableSelectionStatus,
+		selectionMethods,
+	);
+
+	const tableRows = buildTableRows(
+		rows,
+		classes,
+		selectMode,
+		onRowClick,
+		selectionMethods,
+	);
+
+	const stickerTableHeader =
+		stickyHeader === true ? (
+			<StickerTableHeader
+				ref={refHeader}
+				selectMode={selectMode}
+				classes={classes}
+				tableHeaders={tableHeaders}
+			/>
+		) : null;
 
 	return (
-		<Box className={classes.container}>
-			<TableContainer>
-				<TableMui>
-					<TableHead>
-						<TableRow>
-							{checkbox != null ? <TableCellMui>{checkbox}</TableCellMui> : null}
-							{headers.map((header, index) => (
-								<TableCell key={index} cell={header} />
-							))}
-						</TableRow>
-					</TableHead>
-					<TableBody>
-						{rows.map((row, index) => (
-							<TableRow key={index}>
-								{row.map((cell, cellIndex) => (
-									<TableCell key={cellIndex} cell={cell} />
-								))}
-							</TableRow>
-						))}
-					</TableBody>
-				</TableMui>
-				{rows.length > 0 ? null : placeholder}
-			</TableContainer>
-		</Box>
+		<TableContainer className={classes.container}>
+			{stickerTableHeader}
+			<FullTable
+				ref={refScrolled}
+				classes={classes}
+				selectedNumber={selectedNumber}
+				scrollLoader={scrollLoader}
+				tableHeaders={tableHeaders}
+				rows={tableRows}
+				stickyHeader={stickyHeader}
+				tableHeight={tableHeight}
+				contentTimestamp={contentTimestamp}
+				latestPage={latestPage}
+				pageLength={pageLength}
+			/>
+			{tableRows.length > 0 ? null : placeholder}
+		</TableContainer>
 	);
 };
 
-export default Table;
+function tablePropsAreEqual(prev, next) {
+	return (
+		next.contentTimestamp !== undefined && prev.contentTimestamp === next.contentTimestamp
+	);
+}
```

Comment:
I understand that it's easy way to detect changes but, in fact it requires a user to pass an extra parameter which is a bit strange (how timestamp related with table data I want to pass as a user).

It would be great to have a HOC which would add that timestamp for a user. Maybe I don't see the whole picture and there is something what allows final user to not think about that property.

Reply:
An HOC would add a prop that would be managed internally by the HOC, problem here is the user still must provide it to inform the Table component need to render something new, an HOC would not make any differences here. Even though I myself am not sure to like it that solution, I did not find any other way.

Reply:
This question is related to the other in fact, I will resolve this comment and will come to an agreement on the other one.